### PR TITLE
[INLONG-2218][DataProxy] Inlong-DataProxy support authentication access Pulsar

### LINKED ISSUE #2218: [Feature][InLong-DataProxy] Inlong-DataProxy support authentication access pulsar
### Description

Inlong-DataProxy support authentication access  TDMQ

### Use case

Inlong-DataProxy support authentication access  TDMQ

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baomingyu**
Inlong-DataProxy support authentication access TDMQ

Fixes #2218

### CODE REVIEW

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java [hunk 1]
```diff
@@ -228,10 +237,14 @@ private void createConnection(CreatePulsarClientCallBack callBack) throws FlumeE
...
-                .connectionTimeout(clientTimeout, TimeUnit.SECONDS)
-                .build();
+        ClientBuilder builder = PulsarClient.builder();
+        if (PULSAR_DEFAULT_AUTH_TYPE.equals(authType) && StringUtils.isNotEmpty(token)) {
+            builder.authentication(AuthenticationFactory.token(token));
+        }
+        builder.serviceUrl(pulsarUrl)
+                .connectionTimeout(clientTimeout, TimeUnit.SECONDS);
+        return builder.build();
+
```
Comment (dockerzhang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java:247]: remove empty line

File: inlong-dataproxy/conf/flume-mulit-pulsar-udp-example.conf [hunk 2]
```diff
@@ -79,6 +79,9 @@ agent1.channels.ch-msg6.fsyncInterval = 10
 agent1.sinks.pulsar-sink-msg1.channel = ch-msg1
 agent1.sinks.pulsar-sink-msg1.type = org.apache.inlong.dataproxy.sink.PulsarSink
 agent1.sinks.pulsar-sink-msg1.pulsar_server_url_list = pulsar://127.0.0.1:6650
+#Optional pulsar auth type and token config
+#agent1.sinks.pulsar-sink-msg1.pulsar_auth_type = token
+#agent1.sinks.pulsar-sink-msg1.pulsar_token = XXXXXXXXXX
```
Comment (dockerzhang) [inline inlong-dataproxy/conf/flume-mulit-pulsar-udp-example.conf:84]: change the default value for the token，`xxxxx` is not useful.

File: inlong-dataproxy/conf/flume-mulit-pulsar-udp-example.conf [hunk 3]
```diff
@@ -91,6 +94,9 @@ agent1.sinks.pulsar-sink-msg1.disk-io-rate-per-sec=20000000
 agent1.sinks.pulsar-sink-msg2.channel = ch-msg2
 agent1.sinks.pulsar-sink-msg2.type = org.apache.inlong.dataproxy.sink.PulsarSink
 agent1.sinks.pulsar-sink-msg2.pulsar_server_url_list = pulsar://127.0.0.1:6650
+#Optional pulsar auth type and token config
+#agent1.sinks.pulsar-sink-msg2.pulsar_auth_type = token
+#agent1.sinks.pulsar-sink-msg2.pulsar_token = XXXXXXXXXX
```
Comment (dockerzhang) [inline inlong-dataproxy/conf/flume-mulit-pulsar-udp-example.conf:99]: ditto

File: inlong-dataproxy/conf/flume-mulit-pulsar-udp-example.conf [hunk 4]
```diff
@@ -103,6 +109,9 @@ agent1.sinks.pulsar-sink-msg2.disk-io-rate-per-sec=20000000
 agent1.sinks.pulsar-sink-msg5.channel = ch-msg5
 agent1.sinks.pulsar-sink-msg5.type = org.apache.inlong.dataproxy.sink.PulsarSink
 agent1.sinks.pulsar-sink-msg5.pulsar_server_url_list = pulsar://127.0.0.1:6650
+#Optional pulsar auth type and token config
+#agent1.sinks.pulsar-sink-msg5.pulsar_auth_type = token
+#agent1.sinks.pulsar-sink-msg5.pulsar_token = XXXXXXXXXX
```
Comment (dockerzhang) [inline inlong-dataproxy/conf/flume-mulit-pulsar-udp-example.conf:114]: ditto

File: inlong-dataproxy/conf/flume-mulit-pulsar-udp-example.conf [hunk 5]
```diff
@@ -91,6 +94,9 @@ agent1.sinks.pulsar-sink-msg1.disk-io-rate-per-sec=20000000
 agent1.sinks.pulsar-sink-msg2.channel = ch-msg2
 agent1.sinks.pulsar-sink-msg2.type = org.apache.inlong.dataproxy.sink.PulsarSink
 agent1.sinks.pulsar-sink-msg2.pulsar_server_url_list = pulsar://127.0.0.1:6650
+#Optional pulsar auth type and token config
+#agent1.sinks.pulsar-sink-msg2.pulsar_auth_type = token
```
Comment (dockerzhang) [inline inlong-dataproxy/conf/flume-mulit-pulsar-udp-example.conf:98]: add a space after `#`.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java (hunk 1 above)
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/pulsar/PulsarClientService.java]: done

File: inlong-dataproxy/conf/flume-mulit-pulsar-udp-example.conf (hunk 3 above)
Comment (baomingyu) [reply inlong-dataproxy/conf/flume-mulit-pulsar-udp-example.conf]: done

**Review by dockerzhang — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Pull request merged**